lib/spack/docs/contribution_guide.rst

```diff
@@ -316,6 +316,167 @@ documentation tests to make sure there are no errors. Documentation changes can
+    stacks ``spack.yaml``.
+
+
+.. note::
+
+    The platform and architecture components are specified in order to select the
+    correct configurations from the generic configurations used in Spack CI. The
+    predefined configurations currently available are:
+
+    * ``.cray_rhel_zen4``
+    * ``.cray_sles_zen4``
+    * ``.darwin_aarch64``
+    * ``.darwin_x86_64``
+    * ``.linux_aarch64``
+    * ``.linux_icelake``
+    * ``.linux_neoverse_n1``
+    * ``.linux_neoverse_v1``
+    * ``.linux_neoverse_v2``
+    * ``.linux_power``
+    * ``.linux_skylake``
+    * ``.linux_x86_64``
+
+    New configurations can be added to accommodate new platforms and architectures.
+
+
+The build stage is defined as a trigger job that consumes the GitLab CI pipeline generated in
+the generate stage for this stack. Build stage jobs use the ``.build`` job template which
+handles the basic configuration.
+
+
+An example entry point for a new stack called ``my-super-cool-stack``
+
+.. code-block:: yaml
+
+    .my-super-cool-stack:
+      extends: [ ".linux_x86_64_v3" ]
+      variables:
+        SPACK_CI_STACK_NAME: my-super-cool-stack
+        tags: [ "all", "tags", "your", "job", "needs"]
+
+    my-super-cool-stack-generate:
+      extends: [ ".generate", ".my-super-cool-stack" ]
+      image: my-super-cool-stack-image:0.0.1
+
+    my-super-cool-stack-build:
+      extends: [ ".build", ".my-super-cool-stack" ]
+      trigger:
+        include:
+          - artifact: jobs_scratch_dir/cloud-ci-pipeline.yml
+            job: my-super-cool-stack-generate
+        strategy: depend
+      needs:
+        - artifacts: True
+          job: my-super-cool-stack-generate
+
+
+~~~~~~~~~~~~~~~~~~~
+Stack Configuration
+~~~~~~~~~~~~~~~~~~~
+
+The stack configuration is a spack environment file with two additional sections added.
+Stack configurations should be located in ``share/spack/gitlab/cloud_pipelines/stacks/<stack_name>/spack.yaml``.
+
+The first section is optional depending on whether or not the stack will run on runners
+using a docker image or in the runners' native shell, or there are other stack specific
+CI modifications required. For more information on what goes in the ``ci`` section refer to
+the docs on pipelines.
+
+The second section is for CDash. Spack configures most of the details for posting pipeline
+results to `cdash.spack.io <https://cdash.spack.io/index.php?project=Spack+Testing>`_. The only
+requirement in the stack configuration is to define a ``build-group`` that is unique, usually
+this can be a long name of the stack.
+
+An example stack that builds ``zlib``.
+
+.. code-block:: yaml
+
+    spack:
+      view: false
+      packages:
+        all:
+          require: ["%gcc", "target=x86_64_v3"]
+      specs:
+      - zlib
+
+      ci:
+        pipeline-gen
+        - build-job:
+            image: my-super-cool-stack-image:0.0.1
+
+      cdash:
+        build-group: My Super Cool Stack
+
+
+^^^^^^^^^^^^^^^^^^^
+Registering Runners
```

Comment:
It would be nice to document something about how runners can be configured with tags as well, and maybe how the mapping of available jobs to runners ends up behaving.

Especially given the existing stacks are pretty architecture specific, it would be unfortunate if someone added a runner that thought it could run a stack that isn't valid for the runner's architecture.

Reply:
Also, this should maybe document the process for getting a registration key for a runner, what version of the runner is (currently?) supported, and link to a document describing how to install the runner.

Reply:
Thanks for the notes, I will make sure I include these bits in the docs.

I am working on figuring out what the workflow for runner registration should be but we should hopefully have a workflow for that soon. We also need to figure out a way of notifying runner maintainers of breaking upgrades/infrastructure changes so that we can do runner sweep upgrades as well.